src/mscorlib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -957,12 +949,88 @@ internal static String Format(DateTime dateTime, String format, DateTimeFormatIn
             }
 
             if (format.Length == 1) {
+                switch (format[0]) {
+                    case 'o':
```

Comment:
nit: using if-statement maybe better than switch here as we are testing 2 values.